local.Dockerfile

```diff
@@ -1,23 +1,27 @@
 ARG METADATASERVICE_BASE
 ARG SEARCHSERVICE_BASE
 
-FROM node:8 as node-stage
-
-COPY . /app
-
+FROM node:8-slim as node-stage
 WORKDIR /app/amundsen_application/static
 
+COPY amundsen_application/static/package.json /app/amundsen_application/static/package.json
+COPY amundsen_application/static/package-lock.json /app/amundsen_application/static/package-lock.json
 RUN npm install
+
+COPY amundsen_application/static/ /app/amundsen_application/static/
```

Comment:
Can you elaborate on how this speeds things up?

I understanding that copying `package.json` over will optimize `npm-install` when there are no dependency changes during development. However does copying over `static` have the same effect for npm run dev-build? I haven't seen any mention of this in the (limited) amount of time I've spent looking, and curious if  `COPY amundsen_application/static/ /app/amundsen_application/static/` really yields an improvement over having `COPY . /app` at this line

Reply:
Good question!

By limiting what we copy to just the typescript code folder we avoid to trigger npm builds when doing changes to just the Python code.